.github/workflows/on-pr.yml

```diff
@@ -0,0 +1,41 @@
+name: "Validate and plan"
+
+on:
+  pull_request:
```

Comment:
btw, there's already a simple [pull_request worfkflow in this repo](https://github.com/rax-maas/blueflood/blob/master/.github/workflows/unit-test.yml#L6). You'll want to do something with that.